services/pull/pull.go

```diff
@@ -453,6 +453,15 @@ func AddTestPullRequestTask(doer *user_model.User, repoID int64, branch string,
 						}
 					}
 
+					if !pr.IsWorkInProgress(ctx) {
+						reviewNotifiers, err := issue_service.PullRequestCodeOwnersReview(ctx, pr)
```

Comment:
When a CodeOwner has already submitted their review, this will request another review from them, even when there are no new changes to the CodeOwner-protected files. I don't know if that's a problem? A similar kind of issue already existed in the previous implementation: when you toggle the PR's WIP status, it will request another review from the CodeOwner.

Reply:
The condition `isSync` means only commits push will trigger the checking. So that submitting a review comment will not trigger it.

Searching all the `AddTestPullRequestTask`, only the place have `isSync=true`.
<img width="1060" alt="image" src="https://github.com/user-attachments/assets/e18d4eb9-bb5a-413e-ab63-c4f6f45b5776" />

Reply:
What I mean is:

1. person X makes a PR to change file A
2. file A requires a review from person Y via CodeOwners -> review is requested from person Y
3. person Y approves the PR
4. person X pushes a commit to the PR that changes file B
5. file B is not in the CodeOwners, but a new review will still be requested unnecessarily from person Y

And the same unnecessary review request happens if in step 4. instead of pushing a commit, you would mark the PR as WIP and then as ready again.

Reply:
But when the pushed commit does not touch any CodeOwners-protected files, it seems unnecessary.

Reply:
b31d067d72